src/app/server/EnhancedSetupFlowProvider.h

```diff
@@ -0,0 +1,102 @@
+ */
+
+#pragma once
+
+#include <system/SystemConfig.h>
+
+#include <stdint.h>
+
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * @brief Feature state access layer for the EnhancedSetupFlowProvider.
+ *
+ * This class provides access to the state of the Enhanced Setup Flow feature.
+ */
+class EnhancedSetupFlowProvider
+{
+public:
+    /**
+     * @brief Destructor.
+     */
+    virtual ~EnhancedSetupFlowProvider() = default;
+
+    /**
+     * @brief Checks if the required terms and conditions acknowledgements have been accepted.
+     *
+     * @param[out] outAccepted true if the required acknowledgements have been accepted, false otherwise.
+     */
+    virtual CHIP_ERROR HasTermsAndConditionsRequiredAcknowledgementsBeenAccepted(bool & outAccepted) const = 0;
+
+    /**
+     * @brief Checks if the required terms and conditions acknowledgements version has been accepted.
+     *
+     * @param[out] outAccepted true if the required acknowledgements version has been accepted, false otherwise.
+     */
+    virtual CHIP_ERROR HasTermsAndConditionsRequiredAcknowledgementsVersionBeenAccepted(bool & outAccepted) const = 0;
+
+    /**
+     * @brief Retrieves the required terms and conditions acknowledgements.
+     *
+     * @param[out] outValue The version of the required acknowledgements.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsRequiredAcknowledgements(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Retrieves the required terms and conditions acknowledgements version.
+     *
+     * @param[out] outValue The outValue of the required acknowledgements version.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsRequiredAcknowledgementsVersion(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Retrieves the accepted terms and conditions acknowledgements.
+     *
+     * @param[out] outValue The outValue of the accepted acknowledgements.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsAcceptedAcknowledgements(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Retrieves the accepted terms and conditions acknowledgements version.
+     *
+     * @param[out] outValue The outValue of the accepted acknowledgements version.
+     */
+    virtual CHIP_ERROR GetTermsAndConditionsAcceptedAcknowledgementsVersion(uint16_t & outValue) const = 0;
+
+    /**
+     * @brief Sets the acceptance status of the terms and conditions.
+     *
+     * @param[in] inTCAcknowledgements The acknowledgements to accept.
+     * @param[in] inTCAcknowledgementsVersionValue The version of the acknowledgements to accept.
+     */
+    virtual CHIP_ERROR SetTermsAndConditionsAcceptance(uint16_t inTCAcknowledgementsValue,
+                                                       uint16_t inTCAcknowledgementsVersionValue) = 0;
+
+    /**
+     * @brief Clears the acceptance status of the terms and conditions.
+     *
+     */
+    virtual CHIP_ERROR ClearTermsAndConditionsAcceptance() = 0;
```

Comment:
This MUST be called on the removal of the last fabric, otherwise next time the product is getting setup, the prior values of T&C acknowledgements (from a previous user) may be used.

See `OnFabricRemoved` in `Server.h` (`FabricTable::Delegate` impl).